File: by-kotlin/kt-boj/solutions/2012-등수 매기기.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.math.BigDecimal
import kotlin.math.abs

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val n = readLine().toInt()
    val data = mutableListOf<Int>()
    var result = BigDecimal.ZERO

    repeat(n) {
        data.add(readLine().toInt())
    }
    data.sort()

    (1..n).forEach { i ->
        result += abs(i - data[i - 1]).toBigDecimal()
    }

    print(result)
}

File: by-python/boj/2012-등수 매기기.py
import sys

input = sys.stdin.readline


def solve(data):
    data.sort()
    result = 0
    for k in range(n):
        result += abs(data[k] - k - 1)
    return result


if __name__ == '__main__':
    n = int(input())
    data = []
    for _ in range(n):
        data.append(int(input()))

    print(solve(data))
